mm2src/komodefi_cli/src/config.rs

```diff
@@ -58,6 +66,28 @@ pub(super) fn set_config(config: &mut SetConfigArgs) -> Result<()> {
     Ok(())
 }
 
+/// Validates an RPC URL and add base if not provided.
+pub fn validate_rpc_url(input: &str) -> Result<(), anyhow::Error> {
+    let url = match Url::parse(input) {
+        Ok(url) => url,
+        Err(err) => match err {
+            ParseError::RelativeUrlWithoutBase => {
+                Url::parse(&format!("http://{}", input)).map_err(|err| error_anyhow!("Invalid RPC URI: {err:?}"))?
+            },
+            _ => return Err(error_anyhow!("Invalid RPC URI: {err:?}")),
+        },
+    };
+
+    // Check that the scheme is "http" and the host is an IPv4 address, and there is a port
+    if url.scheme() == "http" && url.port().is_some() {
+        Ok(())
+    } else {
+        Err(error_anyhow!(
+            "Invalid RPC URI! Expected scheme, host, and port (http:127.0.0.1:7783)"
+        ))
+    }
+}
```

Comment:
So there are multiple isseus with this:

- I would serve mm2 on the default port, which can be accesed without defining the port in the url.

- I would have the hostname configurated on my machine, so I should be able to access that from any type of url I want.

- I would serve mm2 under tls mode.

- Why is the expected value `http:127.0.0.1:7783` when it's configurable?

Reply:
I guess I didn't make this clear, but what I meant by `http://127.0.0.1:7783` is just telling the user the format that's expected for `uri` so I didn't intend to make this the required value for `url`.

I guess this will be better fit ` "Invalid RPC URI! Expected scheme, host, and port in this format (scheme://host:port)"`

Reply:
Maybe just put `e.g.,` right behind the example value. Also, do you need to put scheme? Can't `127.0.0.1:7783` work alone?

Reply:
okay cool. yes, it accepts host and port only too `127.0.0.1:7783 `

Reply:
I think the scheme validation should be something like:

- check if scheme exists
  - if true, make sure it's either http or https
  - if false, continue
  
And there shouldn't be any validation for the port (we don't know if port will be in the url or not, it depends on the user setup).

So it's only checking the scheme, you shouldn't need to validate the whole url.

I would execute very basic mm2 rpc method(like https://developers.komodoplatform.com/basic-docs/atomicdex-api-legacy/version.html) on the initialization to verify client is correctly settled (I don't know if we already doing this).

Reply:
yeah there's `komodefi_cli mm2 version` for this

and it seems mm2 requires absolute urls for rpc `native_http:174] Invalid request: client requires absolute-form URIs`